alacritty_terminal/src/locale.rs

```diff
@@ -65,24 +65,20 @@ pub fn set_locale_environment() {
         result
     };
 
-    // Provide a default value for LC_CTYPE if needed
+    // provide a default value for LC_CTYPE if needed
     unsafe {
-        let orginal = setlocale(LC_CTYPE, null());
-        let ctype = if orginal.is_null() {
-            CString::new("C").unwrap()
+        // check if we already have LC_CTYPE set as a environment variable
+        if let Ok(ctype) = env::var("LC_CTYPE") {
+            setlocale(LC_CTYPE, CString::new(ctype).unwrap().as_ptr());
         } else {
-            CStr::from_ptr(orginal).to_owned()
+            // if current LC_CTYPE is either null or C, we overwrite it
```

Comment:
The comment isn't actually what we're doing. If the locale is null, we do not set the locale.

Though I don't think that it's possible for a call with a null pointer as parameter to return null, at least according to the documentation. If trying to call it with an invalid locale it will return null, but not set it. So future calls will just return the previous locale.

So I don't think it's necessary to document that specific part.

Reply:
That is now fixed.